Userland/Libraries/LibBencode/Bencode.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2021, Cesar Torres <shortanemoia@protonmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/NonnullOwnPtr.h>
+#include <AK/StdLibExtras.h>
+#include <AK/String.h>
+#include <AK/StringBuilder.h>
+#include <AK/Tuple.h>
+#include <AK/Variant.h>
+#include <AK/Vector.h>
+
+namespace Bencode {
+class BencodeEncoder {
+public:
+    void append(i64 value);
+    void append(StringView const& value);
+    void append(ReadonlyBytes const& value);
+    void start_list();
+    void start_dict();
+    void end_list();
+    void end_dict();
+    NonnullOwnPtr<ByteBuffer> to_buffer();
+    void clear();
+
+private:
+    StringBuilder m_builder {};
+};
+
+enum class BencodeNodeType {
+    Integer,
+    String,
+    List,
+    Dictionary,
+    KeyValuePair
+};
+
+struct BencodeNode;
+
+struct KeyValuePair {
+    String key;
+    NonnullOwnPtr<BencodeNode> value;
+};
+
+struct BencodeNode {
+    BencodeNodeType type;
+    Variant<i64, String, NonnullOwnPtr<KeyValuePair>, NonnullOwnPtr<Vector<NonnullOwnPtr<BencodeNode>>>> data;
+};
+
+Optional<NonnullOwnPtr<BencodeNode>> decode(const ReadonlyBytes& buffer);
+Optional<NonnullOwnPtr<BencodeNode>> decode(const StringView& view);
```

Comment:
How about just using `OwnPtr` instead of `Optional<NonnullOwnPtr`?

Reply:
Done

Reply:
Please click "Resolve Conversation" if you addressed the comment so it gets hidden. Thanks!